meson.build

```diff
@@ -127,19 +125,31 @@ if get_option('omf-byte-order') != 'native'
 endif
 
 add_project_arguments(
-    '-D_GNU_SOURCE',
-    '-D_LGPL_SOURCE',
-    '-Wlogical-op',
-    '-Wno-missing-field-initializers',
-    '-Wmaybe-uninitialized',
-    '-Wno-sign-compare',
-    '-Wno-unused-parameter',
-    '-Wno-clobbered', # TODO: figure out a way to get back to compiling with -Wclobbered (enabled by -Wextra)
-    '-DHSE_LOGPRI_DEFAULT=@0@'.format(log_pri_default),
-    '-DLEVEL1_DCACHE_LINESIZE=@0@'.format(level1_dcache_linesize),
+    cc.get_supported_arguments([
+        '-D_GNU_SOURCE',
+        '-D_LGPL_SOURCE',
+        '-Wlogical-op',
+        '-Wno-missing-field-initializers',
+        '-Wmaybe-uninitialized',
+        '-Wno-sign-compare',
+        '-Wno-unused-parameter',
+        '-DHSE_LOGPRI_DEFAULT=@0@'.format(log_pri_default),
+        '-DLEVEL1_DCACHE_LINESIZE=@0@'.format(level1_dcache_linesize),
+    ]),
     language: 'c',
 )
 
+if cc.get_id() == 'clang'
```

Comment:
Would be nice to eventually get rid of the flags defined in this block.

Reply:
Yeah, especially that variable-sized-type-not-at-end, that's a botch IMO.  Do we have code that does that, or is it something we're pulling in?

Reply:
```
[36/826] Compiling C object lib/libhse-2.so.2.2.0.p/c0_c0_ingest_work.c.o
In file included from ../lib/c0/c0_ingest_work.c:10:
../lib/c0/c0_ingest_work.h:55:5: warning: field '' with variable sized type 'union c0_ingest_work::(anonymous at ../lib/c0/c0_ingest_work.h:55:5)' not at the end of a struct or class is a GNU extension [-Wgnu-variable-sized-type-not-at-end]
    BIN_HEAP2_DEFINE(c0iw_kvms_minheap, HSE_C0_INGEST_WIDTH_MAX);
    ^
../lib/util/include/hse_util/bin_heap.h:57:5: note: expanded from macro 'BIN_HEAP2_DEFINE'
    union {                                                     \
    ^
In file included from ../lib/c0/c0_ingest_work.c:10:
../lib/c0/c0_ingest_work.h:56:5: warning: field '' with variable sized type 'union c0_ingest_work::(anonymous at ../lib/c0/c0_ingest_work.h:56:5)' not at the end of a struct or class is a GNU extension [-Wgnu-variable-sized-type-not-at-end]
    BIN_HEAP2_DEFINE(c0iw_lc_minheap, LC_SOURCE_CNT_MAX);
    ^
../lib/util/include/hse_util/bin_heap.h:57:5: note: expanded from macro 'BIN_HEAP2_DEFINE'
    union {                                                     \
    ^
2 warnings generated.
```